bambooflow/datapipes/aiter.py

```diff
@@ -21,3 +21,56 @@ def __repr__(self) -> str:
         # Instead of showing <bamboopipe. ... .AsyncIterableWrapper at 0x.....>,
         # return the qualified name of the class like <AsyncIterableWrapper>
         return str(self.__class__.__qualname__)
+
+
+class AsyncIterableWrapperAsyncIterDataPipe(AsyncIterDataPipe):
+    """
+    Wraps an iterable object to create an AsyncIterDataPipe.
+
+    Adapted from https://peps.python.org/pep-0492/#example-2
+
+    Parameters
+    ----------
+    iterable : collections.abc.Iterable
+        An :py-term:`iterable` object to be wrapped into an AsyncIterDataPipe.
+
+    Yields
+    ------
+    awaitable : collections.abc.Awaitable
+        An :py-term:`awaitable` object from the
+        :py-term:`asynchronous iterator <asynchronous-iterator>`.
+
+    Example
+    -------
+    >>> import asyncio
+    >>> from bambooflow.datapipes import AsyncIterableWrapper
+    ...
+    >>> # Wrap a list into an asynchronous iterator
+    >>> dp = AsyncIterableWrapper(iterable=[3, 6, 9])
+    ...
+    >>> # Loop or iterate over the DataPipe stream
+    >>> it = aiter(dp)
+    >>> number = anext(it)
+    >>> asyncio.run(number)
+    3
+    >>> number = anext(it)
+    >>> asyncio.run(number)
+    6
+    >>> # Or if running in an interactive REPL with top-level `await` support
+    >>> number = anext(it)  # doctest: +SKIP
+    >>> await number  # doctest: +SKIP
+    9
```

Comment:
Would prefer to use `await number` instead of `asyncio.run(number)` in the doctest, but this would require some very hacky and unmaintainable changes. Possible future solutions might be if pytest markers are supported in doctests per https://github.com/pytest-dev/pytest/issues/5794, then we could use [pytest-asyncio's @pytest.mask.asyncio](https://pytest-asyncio.readthedocs.io/en/latest/reference/markers.html) decorator?

References:
- https://ipython.readthedocs.io/en/8.14.0/interactive/autoawait.html
- https://github.com/python/cpython/pull/13472